File: CurrencyChartObjectiveC/CurrencyChartObjectiveC/ViewController.m
//
//  ViewController.m
//  CurrencyChartObjectiveC
//
//  Created by Clive Alvares on 2017-09-19.
//  Copyright © 2017 Clive Alvares. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    

    // Do any additional setup after loading the view, typically from a nib.
    
    // setting up images
    UIImage *americanFlag = [UIImage imageNamed:@"american-flag-small"];
    UIImage *armFlag = [UIImage imageNamed:@"armenian-flag-small"];
    UIImage *ausFlag = [UIImage imageNamed:@"austrailian-flag-small"];
    UIImage *indianFlag = [UIImage imageNamed:@"indian-flag-small"];
    UIImage *britishFlag = [UIImage imageNamed:@"british-flag-small"];
    UIImage *euFlag = [UIImage imageNamed:@"euFlag"];
    
    // setting up title label
    UILabel *xeLiveEx = [[UILabel alloc] initWithFrame:CGRectMake(11, 20, 183, 45)];
    xeLiveEx.textColor = UIColor.blueColor;
    xeLiveEx.backgroundColor = UIColor.whiteColor;
    xeLiveEx.text = @"XE Live Exchange Rates";
    [xeLiveEx setFont:[UIFont systemFontOfSize:17]];
    [self.view addSubview:xeLiveEx];
    
    [xeLiveEx.leadingAnchor constraintEqualToAnchor: self.view.layoutMarginsGuide.leadingAnchor  constant:0].active = true;
    [xeLiveEx.heightAnchor constraintEqualToConstant:45].active = true;
    [xeLiveEx.widthAnchor constraintEqualToConstant: 183].active = true;
    [xeLiveEx.topAnchor constraintEqualToAnchor:self.view.layoutMarginsGuide.topAnchor constant:10].active = true;
    
    //Set up columns
    
    //set up us column
    UIImageView *amerColImg = [[UIImageView alloc] initWithImage: americanFlag];
    amerColImg.frame = CGRectMake(0, 0, 30, 18.5);
    UIView *amerFlgView = [[UIView alloc] initWithFrame: CGRectMake(0, 27, 30, 18.5)];
    [amerFlgView addSubview:amerColImg];
    
    UILabel *USD = [[UILabel alloc] initWithFrame:CGRectMake(0, 37, 30, 18.5)];
    USD.textColor = UIColor.blackColor;
    USD.text = @"USD";
    UIStackView *colTitle = [[UIStackView alloc] init];
    [colTitle addArrangedSubview:amerFlgView];
    [colTitle addArrangedSubview:USD];
    
    colTitle.axis = UILayoutConstraintAxisVertical;
    colTitle.distribution = UIStackViewDistributionEqualSpacing;
    colTitle.spacing = 0.0;
    colTitle.translatesAutoresizingMaskIntoConstraints = false;
    
    UILabel *usLabel1 = [[UILabel alloc] initWithFrame: CGRectMake(115, 47, 60, 18.5)];
    UILabel *usLabel2 = [[UILabel alloc] initWithFrame: CGRectMake(115, 67, 60, 18.5)];
    UILabel *usLabel3 = [[UILabel alloc] initWithFrame: CGRectMake(115, 77, 60, 18.5)];
    UILabel *usLabel4 = [[UILabel alloc] initWithFrame: CGRectMake(115, 107, 60, 18.5)];
    UILabel *usLabel5 = [[UILabel alloc] initWithFrame: CGRectMake(115, 127, 60, 18.5)];
    UILabel *usLabel6 = [[UILabel alloc] initWithFrame: CGRectMake(115, 147, 60, 18.5)];
    
    usLabel1.text = @"1.00000";
    usLabel2.text = @"1.00000";
    usLabel3.text = @"1.00000";
    usLabel4.text = @"1.00000";
    usLabel5.text = @"1.00000";
    usLabel6.text = @"1.00000";
    
    
    UIStackView *usCol = [[UIStackView alloc] init];
    
    
    usCol.axis = UILayoutConstraintAxisVertical;
    usCol.spacing = 15.0;
    usCol.translatesAutoresizingMaskIntoConstraints = false;
    
    [usCol addArrangedSubview: colTitle];
    [usCol addArrangedSubview: usLabel1];
    [usCol addArrangedSubview: usLabel2];
    [usCol addArrangedSubview: usLabel3];
    [usCol addArrangedSubview: usLabel4];
    [usCol addArrangedSubview: usLabel5];
    [usCol addArrangedSubview: usLabel6];
    
    // set up british col
    UIImageView *britColImg = [[UIImageView alloc] initWithImage: britishFlag];
    britColImg.frame = CGRectMake(0, 0, 30, 18.5);
    UIView *britFlgView = [[UIView alloc] initWithFrame: CGRectMake(0, 27, 30, 18.5)];
    [britFlgView addSubview:britColImg];
    
    UILabel *GBP = [[UILabel alloc] initWithFrame:CGRectMake(0, 37, 30, 18.5)];
    GBP.textColor = UIColor.blackColor;
    GBP.text = @"GBP";
    UIStackView *colBritTitle = [[UIStackView alloc] init];
    [colBritTitle addArrangedSubview:britFlgView];
    [colBritTitle addArrangedSubview:GBP];
    
    colBritTitle.axis = UILayoutConstraintAxisVertical;
    colBritTitle.distribution = UIStackViewDistributionEqualSpacing;
    colBritTitle.spacing = 0.0;
    colBritTitle.translatesAutoresizingMaskIntoConstraints = false;
    
    UILabel *britLabel1 = [[UILabel alloc] initWithFrame: CGRectMake(115, 47, 60, 18.5)];
    UILabel *britLabel2 = [[UILabel alloc] initWithFrame: CGRectMake(115, 67, 60, 18.5)];
    UILabel *britLabel3 = [[UILabel alloc] initWithFrame: CGRectMake(115, 87, 60, 18.5)];
    UILabel *britLabel4 = [[UILabel alloc] initWithFrame: CGRectMake(115, 107, 60, 18.5)];
    UILabel *britLabel5 = [[UILabel alloc] initWithFrame: CGRectMake(115, 127, 60, 18.5)];
    UILabel *britLabel6 = [[UILabel alloc] initWithFrame: CGRectMake(115, 147, 60, 18.5)];
    
    britLabel1.text = @"1.00000";
    britLabel2.text = @"1.00000";
    britLabel3.text = @"1.00000";
    britLabel4.text = @"1.00000";
    britLabel5.text = @"1.00000";
    britLabel6.text = @"1.00000";
    
    
    UIStackView *britCol = [[UIStackView alloc] init];
    
    
    britCol.axis = UILayoutConstraintAxisVertical;
    britCol.spacing = 15.0;
    britCol.translatesAutoresizingMaskIntoConstraints = false;
    
    [britCol addArrangedSubview: colBritTitle];
    [britCol addArrangedSubview: britLabel1];
    [britCol addArrangedSubview: britLabel2];
    [britCol addArrangedSubview: britLabel3];
    [britCol addArrangedSubview: britLabel4];
    [britCol addArrangedSubview: britLabel5];
    [britCol addArrangedSubview: britLabel6];

    // set up armenian column
    UIImageView *armenTitColImg = [[UIImageView alloc] initWithImage: armFlag];
    armenTitColImg.frame = CGRectMake(0, 0, 30, 18.5);
    UIView *armenTitFlgView = [[UIView alloc] initWithFrame: CGRectMake(0, 27, 30, 18.5)];
    [armenTitFlgView addSubview:armenTitColImg];
    
    UILabel *titAMD = [[UILabel alloc] initWithFrame:CGRectMake(0, 37, 30, 18.5)];
    titAMD.textColor = UIColor.blackColor;
    titAMD.text = @"AMD";
    UIStackView *colArmTitle = [[UIStackView alloc] init];
    [colArmTitle addArrangedSubview:armenTitFlgView];
    [colArmTitle addArrangedSubview:titAMD];
    
    colArmTitle.axis = UILayoutConstraintAxisVertical;
    colArmTitle.distribution = UIStackViewDistributionEqualSpacing;
    colArmTitle.spacing = 0.0;
    colArmTitle.translatesAutoresizingMaskIntoConstraints = false;
    
    UILabel *armLabel1 = [[UILabel alloc] initWithFrame: CGRectMake(115, 47, 60, 18.5)];
    UILabel *armLabel2 = [[UILabel alloc] initWithFrame: CGRectMake(115, 67, 60, 18.5)];
    UILabel *armLabel3 = [[UILabel alloc] initWithFrame: CGRectMake(115, 87, 60, 18.5)];
    UILabel *armLabel4 = [[UILabel alloc] initWithFrame: CGRectMake(115, 107, 60, 18.5)];
    UILabel *armLabel5 = [[UILabel alloc] initWithFrame: CGRectMake(115, 127, 60, 18.5)];
    UILabel *armLabel6 = [[UILabel alloc] initWithFrame: CGRectMake(115, 147, 60, 18.5)];
    
    armLabel1.text = @"1.00000";
    armLabel2.text = @"1.00000";
    armLabel3.text = @"1.00000";
    armLabel4.text = @"1.00000";
    armLabel5.text = @"1.00000";
    armLabel6.text = @"1.00000";
    
    
    UIStackView *armCol = [[UIStackView alloc] init];
    
    
    armCol.axis = UILayoutConstraintAxisVertical;
    armCol.spacing = 15.0;
    armCol.translatesAutoresizingMaskIntoConstraints = false;
    
    [armCol addArrangedSubview: colArmTitle];
    [armCol addArrangedSubview: armLabel1];
    [armCol addArrangedSubview: armLabel2];
    [armCol addArrangedSubview: armLabel3];
    [armCol addArrangedSubview: armLabel4];
    [armCol addArrangedSubview: armLabel5];
    [armCol addArrangedSubview: armLabel6];
    
    //set up stack of stacks
    UIStackView *mainStack = [[UIStackView alloc] init];
    [mainStack addArrangedSubview:usCol];
    [mainStack addArrangedSubview:britCol];
    [mainStack addArrangedSubview:armCol];
    mainStack.axis = UILayoutConstraintAxisHorizontal;
    mainStack.distribution = UIStackViewDistributionFillProportionally;
    mainStack.spacing = 0.0;
    mainStack.translatesAutoresizingMaskIntoConstraints = false;
    mainStack.alignment = UIStackViewAlignmentLeading;
    
    [self.view addSubview:mainStack];
    
    
    //set up flag col
    UIImageView *ausColImg = [[UIImageView alloc] initWithImage:ausFlag];
    ausColImg.frame = CGRectMake(0, 0, 30, 18.5);
    UIView *ausFlgView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 30, 18.5)];
    [ausFlgView addSubview:ausColImg];
    UILabel *AUS = [[UILabel alloc] initWithFrame:CGRectMake(30, 20, 60, 18.5)];
    AUS.text = @"AUS";
    UIStackView *ausRow = [[UIStackView alloc] init];
    [ausRow addArrangedSubview:ausFlgView];
    [ausRow addArrangedSubview:AUS];
    ausRow.axis = UILayoutConstraintAxisHorizontal;
    ausRow.distribution = UIStackViewDistributionFillProportionally;
    ausRow.spacing = 0.0;
    ausRow.translatesAutoresizingMaskIntoConstraints = false;
    ausRow.alignment = UIStackViewAlignmentTrailing;
    
    
    UIImageView *armenColImg = [[UIImageView alloc] initWithImage:armFlag];
    armenColImg.frame = CGRectMake(0, 0, 30, 18.5);
    UIView *armenFlgView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 30, 18.5)];
    [armenFlgView addSubview:armenColImg];
    UILabel *AMD = [[UILabel alloc] initWithFrame:CGRectMake(30, 20, 60, 18.5)];
    AMD.text = @"AMD";
    UIStackView *amdRow = [[UIStackView alloc] init];
    [amdRow addArrangedSubview:armenFlgView];
    [amdRow addArrangedSubview:AMD];
    amdRow.axis = UILayoutConstraintAxisHorizontal;
    amdRow.distribution = UIStackViewDistributionFillProportionally;
    amdRow.spacing = 0.0;
    amdRow.translatesAutoresizingMaskIntoConstraints = false;
    amdRow.alignment = UIStackViewAlignmentTrailing;
    
    UIImageView *indColImg = [[UIImageView alloc] initWithImage:indianFlag];
    indColImg.frame = CGRectMake(0, 0, 30, 18.5);
    UIView *indFlgView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 30, 18.5)];
    [indFlgView addSubview:indColImg];
    UILabel *INR = [[UILabel alloc] initWithFrame:CGRectMake(30, 20, 60, 18.5)];
    
    UIStackView *indRow = [[UIStackView alloc] init];
    [indRow addArrangedSubview:indFlgView];
    [indRow addArrangedSubview:INR];
    indRow.axis = UILayoutConstraintAxisHorizontal;
    indRow.distribution = UIStackViewDistributionFillProportionally;
    indRow.spacing = 0.0;
    indRow.translatesAutoresizingMaskIntoConstraints = false;
    indRow.alignment = UIStackViewAlignmentTrailing;
    
    UIStackView *flagCol = [[UIStackView alloc] init];
    [flagCol addArrangedSubview:ausRow];
    [flagCol addArrangedSubview:amdRow];
    [flagCol addArrangedSubview:indRow];
    
    flagCol.axis = UILayoutConstraintAxisVertical;
    flagCol.spacing = 55;
    flagCol.translatesAutoresizingMaskIntoConstraints =false;
    [self.view addSubview:flagCol];
    
    //Column and Row Constraints
    [colTitle.widthAnchor constraintEqualToConstant:30.0].active = true;
    [colTitle.heightAnchor constraintEqualToConstant:37.0].active = true;
    [colBritTitle.widthAnchor constraintEqualToConstant:30].active = true;
    [colBritTitle.heightAnchor constraintEqualToConstant:37].active = true;
    [colArmTitle.widthAnchor constraintEqualToConstant:30].active = true;
    [colArmTitle.heightAnchor constraintEqualToConstant:37].active = true;
    
    [britFlgView.widthAnchor constraintEqualToConstant:30].active = true;
    // ***********************
    [amerFlgView.widthAnchor constraintEqualToConstant:30].active = true;
    [amerFlgView.heightAnchor constraintEqualToConstant:18.5].active = true;
    
    [ausFlgView.widthAnchor constraintEqualToConstant:30].active = true;
    [ausFlgView.heightAnchor constraintEqualToConstant:18.5].active = true;
    
    
    [indFlgView.widthAnchor constraintEqualToConstant:30].active = true;
    [indFlgView.heightAnchor constraintEqualToConstant:18.5].active = true;
    
    [armenFlgView.widthAnchor constraintEqualToConstant:30].active = true;
    [armenFlgView.heightAnchor constraintEqualToConstant:18.5].active = true;
    
    [amerColImg.leadingAnchor constraintEqualToAnchor:amerFlgView.leadingAnchor].active = true;
    [amerColImg.bottomAnchor constraintEqualToAnchor:amerFlgView.bottomAnchor].active = true;
    
    [armenColImg.leadingAnchor constraintEqualToAnchor:armenFlgView.leadingAnchor].active = true;
    [armenColImg.bottomAnchor constraintEqualToAnchor:armenFlgView.bottomAnchor].active = true;
    
    [indColImg.leadingAnchor constraintEqualToAnchor:indFlgView.leadingAnchor].active = true;
    [indColImg.bottomAnchor constraintEqualToAnchor:indFlgView.bottomAnchor].active = true;
    
    [ausColImg.leadingAnchor constraintEqualToAnchor:ausFlgView.leadingAnchor].active = true;
    [ausColImg.bottomAnchor constraintEqualToAnchor:ausFlgView.bottomAnchor].active = true;
    
    [flagCol.topAnchor constraintEqualToAnchor:mainStack.topAnchor constant:50].active = true;
    [flagCol.widthAnchor constraintEqualToConstant:80].active = true;
    [flagCol.leadingAnchor constraintEqualToAnchor:self.view.layoutMarginsGuide.leadingAnchor].active = true;
    [flagCol.heightAnchor constraintEqualToConstant:160].active =true;
    
    
    [mainStack.leadingAnchor constraintEqualToAnchor:flagCol.trailingAnchor].active = true;
    [mainStack.topAnchor constraintEqualToAnchor:xeLiveEx.bottomAnchor constant: 20].active = true;
    [mainStack.heightAnchor constraintEqualToConstant:250].active = true;
    [mainStack.widthAnchor constraintEqualToConstant:320].active = true;
    
    [usCol.heightAnchor constraintEqualToConstant:250].active = true;
    [usCol.widthAnchor constraintEqualToConstant:80].active = true;
    [britCol.heightAnchor constraintEqualToConstant:250].active = true;
    [britCol.widthAnchor constraintEqualToConstant:80].active = true;
    [armCol.heightAnchor constraintEqualToConstant:250].active = true;
    [armCol.widthAnchor constraintEqualToConstant:80].active = true;
    
    
    
    
    
    
    
    
                         
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: CurrencyChartAutolayout/CurrencyChartAutolayout/ViewController.swift
//
//  ViewController.swift
//  CurrencyChartAutolayout
//
//  Created by Clive Alvares on 2017-09-17.
//  Copyright © 2017 Clive Alvares. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        //Set up Title label
        let xeLiveEx = UILabel(frame: CGRect(x:11, y:20, width:183, height:45))
        
        xeLiveEx.textColor = UIColor.blue
        xeLiveEx.backgroundColor = UIColor.white
        xeLiveEx.text = "XE Live Exchange Rates"
        xeLiveEx.font = UIFont.systemFont(ofSize: 17)
        self.view.addSubview(xeLiveEx);
        
        xeLiveEx.leadingAnchor.constraint(equalTo: self.view.layoutMarginsGuide.leadingAnchor, constant: 0).isActive = true
        xeLiveEx.heightAnchor.constraint(equalToConstant: 45).isActive = true
        xeLiveEx.widthAnchor.constraint(equalToConstant: 183).isActive = true
        xeLiveEx.topAnchor.constraint(equalTo: self.view.layoutMarginsGuide.topAnchor, constant: 10).isActive = true
        
        
        //Set up Columns
        // set up us column
        let amerColImg = UIImageView(image: #imageLiteral(resourceName: "american-flag-small"))
        amerColImg.frame = CGRect(x:0, y:0, width:30, height:18.5)
        let amerFlgView = UIView(frame: CGRect(x:0, y:27, width:30, height:18.5))
        amerFlgView.addSubview(amerColImg)
        
        
        
        let USD = UILabel(frame: CGRect(x:0, y:37, width:30, height:18.5))
        USD.textColor = UIColor.black
        USD.text = "USD"
        let colTitle = UIStackView(arrangedSubviews: [amerFlgView, USD])
        colTitle.axis = .vertical
        colTitle.distribution = .equalSpacing
        colTitle.spacing = 0.0
        colTitle.translatesAutoresizingMaskIntoConstraints = false
        
        let usLabel1 = UILabel(frame: CGRect(x:115, y:47, width:60, height:18.5))
        let usLabel2 = UILabel(frame: CGRect(x:115, y:67, width:60, height:18.5))
        let usLabel3 = UILabel(frame: CGRect(x:115, y:87, width:60, height:18.5))
        let usLabel4 = UILabel(frame: CGRect(x:115, y:107, width:60, height:18.5))
        let usLabel5 = UILabel(frame: CGRect(x:115, y:127, width:60, height:18.5))
        let usLabel6 = UILabel(frame: CGRect(x:115, y:147, width:60, height:18.5))
        
        usLabel1.text = "1.00000"
        usLabel2.text = "1.00000"
        usLabel3.text = "1.00000"
        usLabel4.text = "1.00000"
        usLabel5.text = "1.00000"
        usLabel6.text = "1.00000"
        

 
       
        let usCol = UIStackView(arrangedSubviews: [colTitle, usLabel1, usLabel2, usLabel3, usLabel4, usLabel5, usLabel6])
        
        
        usCol.axis = .vertical
        usCol.spacing = 15.0
        usCol.translatesAutoresizingMaskIntoConstraints = false

        
        // set up british col
        let britColImg = UIImageView(image: #imageLiteral(resourceName: "british-flag-small"))
        britColImg.frame = CGRect(x:0, y:0, width:30, height:18.5)
        let britFlgView = UIView(frame: CGRect(x:0, y:27, width:30, height:18.5))
        britFlgView.addSubview(britColImg)
        
        
        
        let GBP = UILabel(frame: CGRect(x:0, y:37, width:30, height:18.5))
        GBP.textColor = UIColor.black
        GBP.text = "GBP"
        let colBritTitle = UIStackView(arrangedSubviews: [britFlgView, GBP])
        colBritTitle.axis = .vertical
        colBritTitle.distribution = .equalSpacing
        colBritTitle.spacing = 20.0
        colBritTitle.translatesAutoresizingMaskIntoConstraints = false
        
        let britLabel1 = UILabel(frame: CGRect(x:115, y:47, width:60, height:18.5))
        let britLabel2 = UILabel(frame: CGRect(x:115, y:67, width:60, height:18.5))
        let britLabel3 = UILabel(frame: CGRect(x:115, y:87, width:60, height:18.5))
        let britLabel4 = UILabel(frame: CGRect(x:115, y:107, width:60, height:18.5))
        let britLabel5 = UILabel(frame: CGRect(x:115, y:127, width:60, height:18.5))
        let britLabel6 = UILabel(frame: CGRect(x:115, y:147, width:60, height:18.5))
        
        britLabel1.text = "1.00000"
        britLabel2.text = "1.00000"
        britLabel3.text = "1.00000"
        britLabel4.text = "1.00000"
        britLabel5.text = "1.00000"
        britLabel6.text = "1.00000"
        
        
        
        
        let britCol = UIStackView(arrangedSubviews: [colBritTitle, britLabel1, britLabel2, britLabel3, britLabel4, britLabel5, britLabel6])
        
        
        britCol.axis = .vertical
        
        britCol.spacing = 15.0
        britCol.translatesAutoresizingMaskIntoConstraints = false
        
        // set up armenian col
        let armenTitColImg = UIImageView(image: #imageLiteral(resourceName: "armenian-flag-small"))
        armenTitColImg.frame = CGRect(x:0, y:0, width:30, height:18.5)
        let armenTitFlgView = UIView(frame: CGRect(x:0, y:27, width:30, height:18.5))
        armenTitFlgView.addSubview(armenTitColImg)
        
        
        
        let titAMD = UILabel(frame: CGRect(x:0, y:37, width:30, height:18.5))
        titAMD.textColor = UIColor.black
        titAMD.text = "AMD"
        let colArmTitle = UIStackView(arrangedSubviews: [armenTitFlgView, titAMD])
        colArmTitle.axis = .vertical
        colArmTitle.distribution = .equalSpacing
        colArmTitle.spacing = 20.0
        colArmTitle.translatesAutoresizingMaskIntoConstraints = false
        
        let armLabel1 = UILabel(frame: CGRect(x:115, y:47, width:60, height:18.5))
        let armLabel2 = UILabel(frame: CGRect(x:115, y:67, width:60, height:18.5))
        let armLabel3 = UILabel(frame: CGRect(x:115, y:87, width:60, height:18.5))
        let armLabel4 = UILabel(frame: CGRect(x:115, y:107, width:60, height:18.5))
        let armLabel5 = UILabel(frame: CGRect(x:115, y:127, width:60, height:18.5))
        let armLabel6 = UILabel(frame: CGRect(x:115, y:147, width:60, height:18.5))
        
        armLabel1.text = "1.00000"
        armLabel2.text = "1.00000"
        armLabel3.text = "1.00000"
        armLabel4.text = "1.00000"
        armLabel5.text = "1.00000"
        armLabel6.text = "1.00000"
        
        
        
        
        let armCol = UIStackView(arrangedSubviews: [colArmTitle, armLabel1, armLabel2, armLabel3, armLabel4, armLabel5, armLabel6])
        
        
        armCol.axis = .vertical
        
        armCol.spacing = 15.0
        armCol.translatesAutoresizingMaskIntoConstraints = false
        // set up stack of stacks
        let mainStack = UIStackView(arrangedSubviews: [usCol, britCol, armCol,])
        mainStack.axis = .horizontal
        mainStack.distribution = .fillProportionally
        mainStack.spacing = 0.0
        mainStack.translatesAutoresizingMaskIntoConstraints = false
        mainStack.alignment = .leading
        
        
        self.view.addSubview(mainStack)
        
        
        //set up flag col
        let ausColImg = UIImageView(image: #imageLiteral(resourceName: "australian-flag-small"))
        ausColImg.frame = CGRect(x:0, y:0, width:30, height:18.5)
        let ausFlgView = UIView(frame: CGRect(x:0, y:0, width:30, height:18.5))
        ausFlgView.addSubview(ausColImg)
        let AUS = UILabel(frame: CGRect(x:30, y:20, width:60, height:18.5))
        AUS.text = "AUS"
        let ausRow = UIStackView(arrangedSubviews: [ausFlgView, AUS])
        ausRow.axis = .horizontal
        ausRow.distribution = .fillProportionally
        ausRow.spacing = 0.0
        ausRow.translatesAutoresizingMaskIntoConstraints = false
        ausRow.alignment = .trailing
        
        
        let armenColImg = UIImageView(image: #imageLiteral(resourceName: "armenian-flag-small"))
        armenColImg.frame = CGRect(x:0, y:0, width:30, height:18.5)
        let armenFlgView = UIView(frame: CGRect(x:0, y:0, width:30, height:18.5))
        armenFlgView.addSubview(armenColImg)
        let AMD = UILabel(frame: CGRect(x:30, y:20, width:60, height:18.5))
        AMD.text = "AMD"
        let amdRow = UIStackView(arrangedSubviews: [armenFlgView, AMD])
        amdRow.axis = .horizontal
        amdRow.distribution = .fillProportionally
        amdRow.spacing = 0.0
        amdRow.translatesAutoresizingMaskIntoConstraints = false
        amdRow.alignment = .trailing
        
        let indColImg = UIImageView(image: #imageLiteral(resourceName: "indian-flag-small"))
        indColImg.frame = CGRect(x:0, y:0, width:30, height:18.5)
        let indFlgView = UIView(frame: CGRect(x:0, y:0, width:30, height:18.5))
        indFlgView.addSubview(indColImg)
        let INR = UILabel(frame: CGRect(x:30, y:20, width:60, height:18.5))
        INR.text = "INR"
        let indRow = UIStackView(arrangedSubviews: [indFlgView, INR])
        indRow.axis = .horizontal
        indRow.distribution = .fillProportionally
        indRow.spacing = 0.0
        indRow.translatesAutoresizingMaskIntoConstraints = false
        indRow.alignment = .trailing
        
        let flagCol = UIStackView(arrangedSubviews: [ausRow, amdRow, indRow])
        
        flagCol.axis = .vertical
        flagCol.spacing = 55
        flagCol.translatesAutoresizingMaskIntoConstraints = false
        self.view.addSubview(flagCol)
        
        
        colTitle.widthAnchor.constraint(equalToConstant: 30).isActive = true
        colTitle.heightAnchor.constraint(equalToConstant: 37).isActive = true
        colBritTitle.widthAnchor.constraint(equalToConstant: 30).isActive = true
        colBritTitle.heightAnchor.constraint(equalToConstant: 37).isActive = true
        colArmTitle.widthAnchor.constraint(equalToConstant: 30).isActive = true
        colArmTitle.heightAnchor.constraint(equalToConstant: 37).isActive = true
     
        britFlgView.widthAnchor.constraint(equalToConstant:30).isActive = true
        amerFlgView.widthAnchor.constraint(equalToConstant:30).isActive = true
        amerFlgView.heightAnchor.constraint(equalToConstant:18.5).isActive = true
        
        ausFlgView.widthAnchor.constraint(equalToConstant:30).isActive = true
        ausFlgView.heightAnchor.constraint(equalToConstant:18.5).isActive = true
        
        indFlgView.widthAnchor.constraint(equalToConstant:30).isActive = true
        indFlgView.heightAnchor.constraint(equalToConstant:18.5).isActive = true
        
        armenFlgView.widthAnchor.constraint(equalToConstant:30).isActive = true
        armenFlgView.heightAnchor.constraint(equalToConstant:18.5).isActive = true
        
        amerColImg.leadingAnchor.constraint(equalTo: amerFlgView.leadingAnchor, constant: 0).isActive = true
        amerColImg.bottomAnchor.constraint(equalTo: amerFlgView.bottomAnchor, constant: 0).isActive = true
        
        armenColImg.leadingAnchor.constraint(equalTo: armenFlgView.leadingAnchor, constant: 0).isActive = true
        armenColImg.bottomAnchor.constraint(equalTo: armenFlgView.bottomAnchor, constant: 0).isActive = true
        
        indColImg.leadingAnchor.constraint(equalTo: indFlgView.leadingAnchor, constant: 0).isActive = true
        indColImg.bottomAnchor.constraint(equalTo: indFlgView.bottomAnchor, constant: 0).isActive = true
        
        ausColImg.leadingAnchor.constraint(equalTo: ausFlgView.leadingAnchor, constant: 0).isActive = true
        ausColImg.bottomAnchor.constraint(equalTo: ausFlgView.bottomAnchor, constant: 0).isActive = true
        
        flagCol.topAnchor.constraint(equalTo: mainStack.topAnchor , constant: 50).isActive = true
        flagCol.widthAnchor.constraint(equalToConstant: 80).isActive = true
        flagCol.leadingAnchor.constraint(equalTo: self.view.layoutMarginsGuide.leadingAnchor, constant: 0).isActive = true
        flagCol.heightAnchor.constraint(equalToConstant: 160.0).isActive = true
        
        
        
        mainStack.leadingAnchor.constraint(equalTo: flagCol.trailingAnchor, constant: 0).isActive = true
        mainStack.topAnchor.constraint(equalTo: xeLiveEx.bottomAnchor, constant: 20).isActive = true
        mainStack.heightAnchor.constraint(equalToConstant: 250.0).isActive = true
        mainStack.widthAnchor.constraint(equalToConstant: 320).isActive = true
        
        usCol.heightAnchor.constraint(equalToConstant: 250.0).isActive = true
        usCol.widthAnchor.constraint(equalToConstant: 80).isActive = true
        britCol.heightAnchor.constraint(equalToConstant: 250.0).isActive = true
        britCol.widthAnchor.constraint(equalToConstant: 80).isActive = true
        armCol.heightAnchor.constraint(equalToConstant: 250.0).isActive = true
        armCol.widthAnchor.constraint(equalToConstant: 80).isActive = true
      
        
        
        
        



       
        
        
        
        
        
        
        
        
        
        
        
        
        
       
        
        
        
        
        
        
    }


    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

